bom/runtime/pom.xml

```diff
@@ -2261,6 +2261,11 @@
                 <artifactId>kotlin-compiler-embeddable</artifactId>
                 <version>${kotlin.version}</version>
             </dependency>
+            <dependency>
+                <groupId>org.jetbrains.kotlin</groupId>
+                <artifactId>kotlin-compiler</artifactId>
+                <version>${kotlin.version}</version>
+            </dependency>
```

Comment:
Do we still need `kotlin-compiler-embeddable`? AFAICS, it's used only in `kotlin/deployment/pom.xml`. Which also raises a question why it is in the runtime BOM instead of the deployment one.

Reply:
Nice catches, thanks! 
I think we can remove `embeddable`. I'll also check and see if moving it to the deployment BOM works.

Reply:
Actually, are you sure it should in the deployment BOM? I don't see any other non-quarkus artifacts thee

Reply:
I removed the embeddable dependency

Reply:
> Actually, are you sure it should in the deployment BOM? I don't see any other non-quarkus artifacts there

The runtime BOM is for runtime dependencies. It is also imported directly by user applications to align application runtime dependencies with the Quarkus platform's runtime dependencies.
Theoretically, if the artifact is only a deployment dependency then it belongs to the deployment BOM. One caveat in the current implementation (and afaict in the to be implemented classloading model) if there are difference versions of the artifact in the runtime CP and the deployment CP then the runtime one will end up being used during the augmentation.
In this given case, however, I'd probably put it in the deployment BOM. If you are not comfortable doing that then don't :) Could be a separate issue.
The BOMs should probably go through a thorough review.

Reply:
I did a test that moved the dependency to the deployment BOM and as you expected it worked. The only caveat however is that we need to redeclare the kotlin version and keep it in sync with the runtime BOM.
Should we go for that or keep it as it is?

Reply:
Ah, the properties. I think the properties should move to the root pom.xml so that they can be shared across all the subprojects including the tests. Some tests define/duplicate those versions, which is not a good thing.

Reply:
@aloubyansky makes sense!

@gsmet I think that what @aloubyansky proposes should be handled in a dedicated issue, WDYT?

Reply:
I fully agree about having the properties in their own pom. I suggested that a while ago.

Not entirely sure it will work to have them in the root pom, depends if all the pom requiring them has it as parent. Otherwise, it can be a specific pom.

I agree it should be done in a separate issue.

Reply:
Thanks both of you for your input!

Reply:
> Not entirely sure it will work to have them in the root pom, depends if all the pom requiring them has it as parent. Otherwise, it can be a specific pom.

Either way that properties pom will have to be a parent pom for every subproject. Which is what the root pom is supposed to be, otherwise the hierarchy for the project is going to break which effectively means independent projects.